azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,406 @@
+			},
+
+			//"backup_blob_container_uri": {
+			//	Type:         schema.TypeString,
+			//	Optional:     true,
+			//	Sensitive:    true,
+			//	ValidateFunc: validate.URLIsHTTPOrHTTPS,
+			//},
+
+			"enable_power_bi_service": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"ipv4_firewall_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"range_start": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.IPv4Address,
+						},
+						"range_end": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.IPv4Address,
+						},
+					},
+				},
+			},
+
+			"querypool_connection_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateQuerypoolConnectionMode(),
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAnalysisServicesServerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		server, err := client.GetDetails(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(server.Response) {
+				return fmt.Errorf("Error checking for presence of existing Analysis Services Server %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if server.ID != nil && *server.ID != "" {
+			return tf.ImportAsExistsError("azurerm_analysis_services_server", *server.ID)
+		}
+	}
+
+	sku := d.Get("sku").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+
+	serverProperties := expandAnalysisServicesServerProperties(d)
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	analysisServicesServer := analysisservices.Server{
+		Name:             &name,
+		Location:         &location,
+		Sku:              &analysisservices.ResourceSku{Name: &sku},
+		ServerProperties: serverProperties,
+		Tags:             expandTags(tags),
+	}
+
+	future, err := client.Create(ctx, resourceGroup, name, analysisServicesServer)
+	if err != nil {
+		return fmt.Errorf("Error creating Analysis Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for completion of Analysis Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	server, err := client.GetDetails(ctx, resourceGroup, name)
+	if err != nil {
+		return err
```

Comment:
to make this more descriptive for users - could we wrap this error?

```suggestion
		return fmt.Errorf("Error retrieving Analytics Services Server %q (Resource Group %q): %+v", name, resourceGroup, err)
```